src/components/Articles.jsx

```diff
@@ -6,38 +6,49 @@ export default function Articles() {
 
   const {topic_id} = useParams()
 
-  console.log(topic_id)
-
   const [articleList, setArticleList] = useState([])
   const [pageNumber, setPageNumber] = useState(1)
   const [totalArticles, setTotalArticles] = useState(0)
   const [isFirstPage, setIsFirstPage] = useState(true)
   const [isLastPage, setIsLastPage] = useState(false)
+  const [sortByDate, setSortByDate] = useState(false)
+  const [sortByCommentCount, setSortByCommentCount] = useState(false)
+  const [sortByVotes, setSortByVotes] = useState(false)
+  const [isAscending, setIsAscending] = useState(false)
 
   const pageLimit = Math.ceil(totalArticles / 10)
 
+  let urlString = `/articles?`
+
+  if (topic_id) {
+    urlString+= `topic=${topic_id}&`
+  }
+
+  if (sortByDate) {
+    urlString+= `sort_by=created_at&`
+  }
+
+  if (sortByCommentCount) {
+    urlString+= `sort_by=comment_count&`
+  }
+
+  if (sortByVotes) {
+    urlString+= `sort_by=votes&`
+  }
+
+  if (isAscending) {
+    urlString+= `order=asc&`
+  }
+
+  urlString+= `p=${pageNumber}`
+
   useEffect(() => {
-    if (!topic_id)
-    newsApi.get(`/articles?p=${pageNumber}`)
+    newsApi.get(urlString)
```

Comment:
this can 100% be refactored to use axios.params. that will help you clear out all the state you have as well